install_config/advanced_ldap_configuration/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,425 @@
+use a Red Hat Enterprise Linux 7.2 Server.
+
+. Install all of the necessary dependencies:
++
+====
+----
+# yum install -y sssd \
+                 sssd-dbus \
+                 realmd \
+                 httpd \
+                 mod_session \
+                 mod_ssl \
+                 mod_lookup_identity \
+                 mod_authnz_pam
+----
+====
++
+This gives you the needed SSSD and the web server components.
+
+. Set up SSSD to authenticate this VM against the LDAP server. If the LDAP server
+is a FreeIPA or Active Directory environment, then *realmd* can be used to join
+this machine to the domain.
++
+====
+----
+# realm join ldap.example.com
+----
+====
++
+If you are not running a domain, then use the *authconfig* tool:
++
+====
+----
+# authconfig --update --enablesssd --enablesssdauth \
+             --ldapserver=ldap.example.com \
+             --enableldaptls \
+             --ldaploadcert=http://ldap.example.com/ca.crt
+----
+====
++
+This creates *_/etc/sssd/sssd.conf_* with most of the appropriate settings.
++
+[NOTE]
+====
+In RHEL 7, *authconfig* does not create the *_/etc/openldap/cacerts_* directory,
+so you may need to create it manually before running the above command.
+====
++
+If you want to use SSSD to manage failover situations for LDAP, this can be
+configured by adding additional entries in *_/etc/sssd/sssd.conf_* on the
+*ldap_uri* line. Systems enrolled with FreeIPA can automatically handle
+failover using DNS SRV records.
+
+. Restart SSSD to ensure that all of the changes are applied properly:
++
+====
+----
+$ systemctl restart sssd.service
+----
+====
+
+. Test that the user information can be retrieved properly:
++
+====
+----
+$ getent passwd <username>
+username:*:12345:12345:Example User:/home/username:/usr/bin/bash
+----
+====
+
+. Attempt to log into the VM as an LDAP user and confirm that the authentication
+is properly set up. This can be done via the local console or a remote service
+such as SSH.
+
+[[phase-2-step-3-apache-configuration]]
+=== Step 3: Apache Configuration
+
+You need to set up Apache to communicate with SSSD. Create a PAM stack file for
+use with Apache. To do so:
+
+.  Create the *_/etc/pam.d/openshift_* file and add the
+following contents:
++
+====
+----
+auth required pam_sss.so
+account required pam_sss.so
+----
+====
++
+This configuration enables PAM (the pluggable authentication module) to use
+*pam_sss.so* to determine authentication and access control when an
+authentication request is issued for the *openshift* stack.
+
+. Configure the Apache *_httpd.conf_*. The steps in this section focus on
+setting up the challenge authentication, which is useful for logging in with `oc
+login` and similar automated tools.
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how you can get a template login screen.
+====
```

Comment:
The way this is written feels out of place. As I was reading that, I would have thought I should jump over to that page and then come back. Of course, that other page relies on this one being complete, so maybe rephrase to:

"We will discuss in a later chapter[link] how to set up a graphical login using SSSD as well, but it requires the rest of this setup as a prerequisite."

Well, that's really awkward to read, but I'm sure you're a better wordsmith than I am.